pennylane/transforms/__init__.py

```diff
@@ -1,66 +1,80 @@
-
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-"""
-This subpackage contains QNode, quantum function, device, and tape transforms.
-
-
-.. currentmodule:: pennylane
-
-QNode transforms
-----------------
-
-The following transforms act on QNodes. They return new transformed functions
-that compute the desired quantity.
-
-.. autosummary::
-    :toctree: api
-
-    ~transforms.classical_jacobian
-    ~draw
-    ~metric_tensor
-
-Quantum function transforms
----------------------------
-
-The following transforms act on quantum functions (Python functions
-containing quantum operations) that are used *inside* QNodes.
-
-.. autosummary::
-    :toctree: api
-
-    ~adjoint
-    ~ctrl
-    ~transforms.invisible
-
-Tape transforms
----------------
-
-The following transforms act on quantum tapes, and return one or
-more tapes as well as a classical processing function.
-
-.. autosummary::
-    :toctree: api
-
-    ~transforms.measurement_grouping
-    ~transforms.metric_tensor_tape
-    ~transforms.hamiltonian_expand
-"""
-from .adjoint import adjoint
-from .classical_jacobian import classical_jacobian
-from .control import ControlledOperation, ctrl
-from .draw import draw
-from .invisible import invisible
-from .measurement_grouping import measurement_grouping
-from .metric_tensor import metric_tensor, metric_tensor_tape
-from .hamiltonian_expand import hamiltonian_expand
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This subpackage contains QNode, quantum function, device, and tape transforms.
+
+
+.. currentmodule:: pennylane
+
+QNode transforms
+----------------
+
+The following transforms act on QNodes. They return new transformed functions
+that compute the desired quantity.
+
+.. autosummary::
+    :toctree: api
+
+    ~transforms.classical_jacobian
+    ~draw
+    ~metric_tensor
+
+Quantum function transforms
+---------------------------
+
+The following transforms act on quantum functions (Python functions
+containing quantum operations) that are used *inside* QNodes.
+
+.. autosummary::
+    :toctree: api
+
+    ~adjoint
+    ~ctrl
+    ~transforms.invisible
+
+Tape transforms

```

Comment:
Do these contents need to be updated to fit the new description of "tape transform", or will that be done as part of #1362?

Reply:
Oh good question. I think I have been avoiding updating the docs for now, since this is so fluid....

Reply:
I think that tape transforms themselves are in a reasonably final form, with the only factor left being the multiple tape case. That said, since the focus and decorator here is `single_tape_transform` we could go with that as the definition of tape transform for now, but  leave open the potential to later include `multi_tape_transform` (or something like this) as an additional part of the definition.

Reply:
Though I guess if we did that we would have to move the three transforms in the autosummary to a different section :confused:

Reply:
I'm kinda tempted to leave it very loosely defined for now, to avoid locking us in to anything too early. Instead:

- a tape transform is simply a function that maps tape(s) to tape(s). Any custom function can be written that does this.
- `@qml.single_tape_transform` provides an API for easily creating a particular _type_ of tape transform that is restricted in some snse.

That is, rather than `@qml.single_tape_transform` or some future `@qml.multi_tape_transform` **defining** what a tape transform is, they simply provide accessible ways to easily create two restricted types of tape transforms.

Reply:
Nice, okay I'm on board with that!